pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -350,14 +394,44 @@ class CreateVM extends React.Component {
                                  onValueChange={this.onChangedEventValue.bind(this, 'memorySize')}
                                  onUnitChange={this.onChangedValue.bind(this, 'memorySizeUnit')} />
 
-                {this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
+                { this.state.storagePool === "Automatic" &&
+                this.state.sourceType != EXISTING_DISK_IMAGE_SOURCE &&
                 <MemorySelectRow label={_("Storage Size")}
                                  id={"storage-size"}
                                  value={this.state.storageSize}
                                  initialUnit={this.state.storageSizeUnit}
                                  onValueChange={this.onChangedEventValue.bind(this, 'storageSize')}
                                  onUnitChange={this.onChangedValue.bind(this, 'storageSizeUnit')} />}
 
+                <label className="control-label" htmlFor="storage-pool">
```

Comment:
The whole storage pool, volume, size section should not be visible when the user selects to use existing disk image.

Reply:
Also I tend to believe that the storage size, storage pool, storage volume fields should get separated from the memory field, but that's up to @garrett  to decide.